clustered/server/src/main/java/org/ehcache/clustered/server/EhcacheActiveEntity.java

```diff
@@ -241,13 +262,40 @@ public EhcacheEntityResponse invoke(ClientDescriptor clientDescriptor, EhcacheEn
                                                   " Check your server configuration and define at least one offheap resource.");
       }
 
+      //All the invokes are blocked till passive evicted keys are invalidated by server under lock
+      //Once invalidations are done, other threads will just move on by checking the status
+      if (!invalidationForPassiveEvictedKeysSent.get()) {
```

Comment:
Let's ditch all that logic.

It is not wrong per caching contract to have clients return null while others return the valid mapping. In all likelihood, the first client seeing null will want to update and that will be correctly propagated to all clients.

And in addition, I am really not fond of that locking done here.